cloud-benchmark/azure/terraform/main.tf

```diff
@@ -183,108 +182,153 @@ resource "azurerm_container_app" "cloud_benchmark_single_node" {
-      }
-
-      env {
-        name  = "AUCTIONMARK_LOAD_PHASE"
-        value = true
-      }
-
-      env {
-        name  = "AUCTIONMARK_LOAD_PHASE_ONLY"
-        value = false
-      }
+  oidc_issuer_enabled       = true
+  workload_identity_enabled = true
+}
 
-      env {
-        name  = "XTDB_LOCAL_DISK_CACHE"
-        value = "/var/lib/xtdb/disk-cache/cache-1"
-      }
+resource "azurerm_role_assignment" "cloud_benchmark_cluster_role" {
+  count                            = var.run_single_node ? 1 : 0
+  principal_id                     = azurerm_kubernetes_cluster.cloud_benchmark[0].kubelet_identity[0].object_id
+  role_definition_name             = "AcrPull"
+  scope                            = azurerm_container_registry.acr.id
+  skip_service_principal_aad_check = true
+}
 
-      env {
-        name  = "XTDB_AZURE_USER_MANAGED_IDENTITY_CLIENT_ID"
-        value = azurerm_user_assigned_identity.cloud_benchmark.client_id
-      }
+resource "azurerm_federated_identity_credential" "cloud_benchmark" {
+  count               = var.run_single_node ? 1 : 0
+  name                = "cloud-benchmark-worker"
+  resource_group_name = azurerm_resource_group.cloud_benchmark.name
+  audience            = ["api://AzureADTokenExchange"]
+  issuer              = azurerm_kubernetes_cluster.cloud_benchmark[0].oidc_issuer_url
+  parent_id           = azurerm_user_assigned_identity.cloud_benchmark.id
+  subject             = "system:serviceaccount:${local.namespace}:${local.service_account_name}"
+}
 
-      env {
-        name  = "XTDB_AZURE_STORAGE_ACCOUNT"
-        value = azurerm_storage_account.cloud_benchmark.name
-      }
+resource "local_file" "kubeconfig" {
+  count      = var.run_single_node ? 1 : 0
+  depends_on = [azurerm_kubernetes_cluster.cloud_benchmark]
+  filename   = "kubeconfig"
+  content    = azurerm_kubernetes_cluster.cloud_benchmark[0].kube_config_raw
+}
 
-      env {
-        name  = "XTDB_AZURE_STORAGE_CONTAINER"
-        value = azurerm_storage_container.cloud_benchmark.name
-      }
+provider "kubernetes" {
+  config_path = local_file.kubeconfig[0].filename
+}
 
-      env {
-        name  = "XTDB_AZURE_SERVICE_BUS_NAMESPACE"
-        value = azurerm_servicebus_namespace.cloud_benchmark.name
-      }
+resource "kubernetes_namespace" "cloud_benchmark" {
+  count = var.run_single_node ? 1 : 0
+  metadata {
+    name = local.namespace
+  }
+}
 
-      env {
-        name  = "XTDB_AZURE_SERVICE_BUS_TOPIC_NAME"
-        value = azurerm_servicebus_topic.cloud_benchmark.name
-      }
+resource "kubernetes_service_account" "cloud_benchmark" {
+  count = var.run_single_node ? 1 : 0
+  metadata {
+    name      = local.service_account_name
+    namespace = kubernetes_namespace.cloud_benchmark[0].metadata[0].name
+  }
+}
 
-      env {
-        name  = "XTDB_AZURE_EVENTHUB_NAMESPACE"
-        value = azurerm_eventhub_namespace.cloud_benchmark.name
-      }
+resource "kubernetes_deployment" "cloud_benchmark" {
```

Comment:
All looking pretty sensible to me so far :)

I was keen for us to try setting up a persistent volume (might be worth a look at some of the google-cloud kubernetes files) given that was one of our big reasons to switching over to AKS.